packages/core/test-helpers/core-test-helpers-so-type-serializer/src/get_migration_hash.ts

```diff
@@ -26,8 +26,19 @@ export const getMigrationHash = (soType: SavedObjectsType): SavedObjectTypeMigra
     migInfo.migrationVersions.join(','),
     migInfo.schemaVersions.join(','),
     JSON.stringify(migInfo.mappings, Object.keys(migInfo.mappings).sort()),
+    migInfo.switchToModelVersionAfter ?? 'none',
+    migInfo.modelVersions.map(serializeModelVersion).join(','),
   ];
   const hashFeed = hashParts.join('-');
 
   return hash.update(hashFeed).digest('hex');
 };
+
+const serializeModelVersion = (modelVersion: ModelVersionSummary): string => {
+  return [
+    modelVersion.version,
+    modelVersion.changeType,
+    modelVersion.hasMigration,
+    ...modelVersion.newMappings,
+  ].join('|');
```

Comment:
I'm reading the code, and we only serialize the model version and never require to deserialize it.
However, if we ever needed that we will face issues if we are not restricting field names in the mappings to contain `|` or escaping them during serialization.

Do you think we'll ever need that functionality or should we treat this solely as part of the one way hashing function?

Reply:
For now, the hashing is only required for this 'did the registered types change' logic, for for now we're good.

If we need full serialization/deserialization for, say, the migration algo, we would need some smarter hashing, but I don't think we'll ever need it, so we should be ok.